scripts/build_jacocorunner/build_jacocorunner.sh

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env bash
+#
+# Script to build custom version of `JacocoCoverage_jarjar_deploy.jar` from Jacoco and Bazel repositories.
+#
+# The default `JacocoCoverage_jarjar_deploy.jar` has some issues:
+#
+# 1. Bazel uses Jacoco 0.8.3 that has poor Scala support, including a bug that filters out Scala lambdas on Scala >=2.12
+#
+#    Bug report:
+#    https://github.com/bazelbuild/rules_scala/issues/1056
+#    https://github.com/bazelbuild/bazel/issues/11674
+#
+#    Backported fix from Jacoco 0.8.5 to Jacoco 0.8.3 (current Bazel is not compatible with Jacoco 0.8.5):
+#    https://github.com/gergelyfabian/jacoco/tree/0.8.3-scala-lambda-fix
+#
+# 2. Bazel ignores Jacoco's filtering for branch coverage metrics:
+#
+#    Bug report:
+#    https://github.com/bazelbuild/bazel/issues/12696
+#
+#    Proposed fix:
+#    https://github.com/gergelyfabian/bazel/tree/branch_coverage_respect_jacoco_filtering
+#
+# 3. Scala support on newer Jacoco versions (including 0.8.5) is still lacking some functionality
+#
+#    E.g. a lot of generated methods for case classes, lazy vals or other Scala features are causing falsely missed branches in branch coverage.
+#
+#    Proposed changes in:
+#    https://github.com/gergelyfabian/jacoco/tree/scala
+#
+#    Backported to 0.8.3 (to be usable with current Bazel):
+#    https://github.com/gergelyfabian/jacoco/tree/0.8.3-scala
+#
+#    These branches also include the Scala 2.12 lambda coverage fix.
+#
+# You can use this script to build a custom version of `JacocoCoverage_jarjar_deploy.jar`, including any fixes from the above list you wish
+# and then provide the built jar as a parameter of `java_toolchain` and/or `scala_toolchain` to use the changed behavior for coverage.
+#
+# Choose the fixes from the above list by configuring the used branches for Bazel and Jacoco repos below.
+#
+# Patches:
+#
+# There are also some patches that may need to be applied for Jacoco, according to your preferences:
+#
+# 1. Bazel is compatible only with Jacoco in a specific package name. This is not Jacoco-specific, so not committed to the Jacoco fork.
+#    See 0001-Build-Jacoco-for-Bazel.patch.
+# 2. Building Jacoco behind a proxy needs a workaround.
+#    See 0002-Build-Jacoco-behind-proxy.patch.
+#
+# Set up the patch file names in `jacoco_patches`.
+#
+# Dependencies:
+#
+# On OS X greadlink is needed (by running `brew install coreutils`).
+
+set -e
+
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+  source_path=$(readlink -f $(dirname ${BASH_SOURCE[0]}))
```

Comment:
Maybe only have the process name be part of the if? This way we have less duplication and less chance of errors later

Reply:
Done.